nautobot/docs/development/apps/api/views/nautobotuiviewset.md

```diff
@@ -145,3 +145,53 @@ For example, for a DetailView template for `YourAppModel`, the template name wil
 If you do not provide your own templates in the `yourapp/templates/yourapp` folder, `NautobotUIViewSet` will fall back to `generic/object_{self.action}.html`.
 
 Since in many cases the `create` and `update` templates for a model will be identical, you are not required to create both. If you provide a `{app_label}/{model_opts.model_name}_create.html` file but not a `{app_label}/{model_opts.model_name}_update.html` file, then when you update an object, it will fall back to `{app_label}/{model_opts.model_name}_create.html` and vice versa.
+
+### Adding Custom Views To NautobotUIViewSet & NautobotUIViewSetRouter
+
++++ 1.6.0
+    Via [PR #4045](https://github.com/nautobot/nautobot/pull/4045), notes and changelog views provided by mixins have now been moved to this pattern.
+
+Django REST Framework provides the ability to decorate a method on a ViewSet with `@action(detail=True)` to add the method as a view to the ViewSetRouter. This method must return a fully rendered HTML view.
+
+Below is an example of adding a custom action view to the plugin ViewSet. A few considerations to keep in mind:
+
+* The method name is the `action` in Django REST Framework terms.
+* The `action` will be used for [template lookup](#template-naming-for-nautobotuiviewset)
+* The `action` will be used for URL naming and construction (`plugins:<app>:<model>_<action>`, `/plugins/<app>/<model>/<uuid>/<action>/`).
+* The `action` will be used as a custom permission that users must have (`<app>.<action>_<model>`)
+
+In the below example:
+
+* The expected template must be named `yourapp/yourappmodel_customview.html`
+* The reversible URL name will be `plugins:yourapp:yourappmodel_customview`
+* The URL pattern will be `/plugins/yourapp/yourappmodel/<uuid>/customview/`
+* Users will need `yourapp.customview_yourappmodel` permission to access this view.
```

Comment:
Instead of `yourappmodel_customview` would it be more clear to call this `yourappmodel_customaction`?

Reply:
Maybe not, now that I've read this whole doc